### PR TITLE
App block

### PR DESCRIPTION
**Pull request by mcandeia**

### CODE REVIEW

File: engine/schema/utils.ts [hunk 1]
```diff
@@ -167,11 +161,20 @@ function isQuotaExceededError(err: unknown): boolean {
...
   path: string,
-  importMap?: string,
+  _importMap?: string,
 ): Promise<DocNode[]> => {
   try {
+    const docCacheKey = path;
+    if (context.isDeploy) {
+      const cached = denoDocLocalCache[docCacheKey];
+      if (!cached) {
+        throw new Error(`could not resolve ${docCacheKey} on denodoc`);
```
Comment (matheusgr) [inline engine/schema/utils.ts:174]: Offer a solution for the probably error cause.

```suggestion
        throw new Error(`could not resolve ${docCacheKey} on denodoc`); Create denodoc cache locally with "deno task doccache" and push the denocache.zst file.
```

Comment (matheusgr) [reply engine/schema/utils.ts:174]: Do we have a task to create the deno doc cache?

File: blocks/app.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  TAppManifest extends AppManifest = AppManifest,
+  // deno-lint-ignore ban-types
+  TAppState = {},
+> {
+  manifest: TAppManifest;
+  state: TAppState;
+}
+
+export type AppContext<TApp extends App = App> = FnContext<
+  TApp["state"],
```
Comment (matheusgr) [inline blocks/app.ts:32]: Maybe `TApp["state"]` should merge its own state and those from its dependencies?

Something like that may be better modeled as Record<namespace, {}>.

**Review by matheusgr — CHANGES_REQUESTED** [review]

File: engine/schema/utils.ts (hunk 1 above)
Comment (mcandeia) [reply engine/schema/utils.ts:174]: yes. deno task start --gen-only

File: blocks/app.ts (hunk 2 above)
Comment (mcandeia) [reply blocks/app.ts:32]: Discussed offline

File: scripts/apps/init.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+const templates: Templates = ["import_map.json", "deno.json", "deco.ts", "state.ts", "deps.ts", {
+  app: {
+    loaders: ["bin.ts"],
+  },
+}];
+
+const init = async () => {
+  const latestVersionPromise = lookup(
+    "https://denopkg.com/deco-cx/deco@main/",
```
Comment (matheusgr) [inline scripts/apps/init.ts:73]: Don't see main as a common version name/tag.

```suggestion
    "https://denopkg.com/deco-cx/deco@latest/",
```

**Review by matheusgr — APPROVED** [review]

**Pull request merged**